_blog/deploying-laravel-to-kubernetes/index.md

```diff
@@ -0,0 +1,354 @@
+FROM php:7
+RUN apt-get update -y && apt-get install -y openssl zip unzip git
+RUN curl -sS https://getcomposer.org/installer | php -- --install-dir=/usr/local/bin --filename=composer
+RUN docker-php-ext-install pdo mbstring
+WORKDIR /app
+COPY . /app
+RUN composer install
+
+CMD php artisan key:generate
+CMD php artisan serve --host=0.0.0.0 --port=8181
+EXPOSE 8181
+``` 
+
+It extends a PHP base image, installs some system dependencies including 
+composer and the standard PHP extensions required by Laravel. It will then
+ copy the application files to a working directory and install the application's 
+  dependencies via composer. It also runs a couple of `artisan` commands, 
+  last of which serves the application using the built-in PHP web server. At the
+   end, it exposes port `8181` to the deployment.
+  
+To build the local Docker image:
+
+```bash
+cd /to/your/project/directory
+docker build -t yourname/laravel-kubernetes-demo .
+```
+
+__Deploying the image__
+
+Now that the application's image is built and is available for deploying it 
+locally, let's go ahead and deploy it.
+
+I always start with making sure that Kubectl is in the right context. In our 
+case, the context is Minikube. We can simply switch context as follows:
+
+```bash
+kubectl config use-context minikube
+```
+
+then we can deploy the container image:
+
+```bash
+kubectl run laravel-kubernetes-demo --image=yourname/laravel-kubernetes-demo 
+--port=8181 --image-pull-policy=IfNotPresent
+```
+
+The above command tells Kubectl to run our demo application from the Docker 
+image, whilst making port 8181 available for listening. The last parameter of 
+the command simply asks Kubectl to not pull the image from a registry such as
+ Docker Hub if it exists locally which in our case it does. Do note that you 
+ still need to be logged on to Docker so that Kubectl can check if the image 
+ is up to date.
+ 
+We can check that a Pod is created for our application by running:
+
+```bash
+kubectl get pods
+```
+
+which should return a similar output to:
+
+```bash
+NAME                                       READY     STATUS    RESTARTS   AGE
+laravel-kubernetes-demo-7dbb9d6b48-q54wp   1/1       Running   0          18m
+```
+
+We can also use the Minikube GUI dashboard to monitor our cluster. The GUI 
+also helps with visualising most of the discussed concepts. To view the 
+dashboard, simply run the following:
+
+```bash
+minikube dashboard
+```
+
+or to acquire the dashboard's URL address:
+
+```bash
+minikube dashboard --url=true
+```
+
+which will return the IP address and the port number of the dashboard.
+
+__Exposing a service__
+
+So far we have created a deployment which is running our application's 
+container. With the container running, we need to expose the PHP's built-in web 
+server as a service.
+
+```bash
+kubectl expose deployment laravel-kubernetes-demo --type=NodePort --port=8181
+```
+
+and provided all went well, we should see a confirmation similar to:
```

Comment:
I suggest turning this into something similar to the pods command earlier:

```bash
$ kubectl get services
```

If you run the command, you can show that the service has an IP address assigned.

Reply:
I excluded the `kubectl get services` command so that the reader does not get confused with the internal IP address. I'll add it anyway.